docs/Using-Fleet/MDM-macOS-setup.md

```diff
@@ -0,0 +1,120 @@
+# macOS setup
+
+In Fleet, you can customize the first-time macOS setup experience for your end users:
+
+You can customize the macOS Setup Assistant and choose to show or hide specific panes.
+* Require end users to authenticate with your identity provider (IdP) and agree to an end user license agreement (EULA) before they can use their new Mac
+
+* Customize the macOS Setup Assistant by choosing to show or hide specific panes
+
+* Bootstrap new Macs with your configuration management tool of choice (ex. Munki, Chef, or Puppet)
+
+## End user authentication
+
+> This feature is currently in development.
+
+## Bootstrap package
+
+_Available in Fleet Premium_
+
+Fleet supports installing a single package on macOS hosts that automatically enroll to Fleet.
+
+> In addition to installing the bootstrap package, Fleet automatically installs the fleetd agent on hosts that automatically enroll. This agent is responsible for reporting host vitals to Fleet and presenting Fleet Desktop to the end user.
+
+To add a bootstrap package to Fleet, we will do the following steps:
+
+1. Download or generate a package
+2. Sign the package
+3. Upload the package to Fleet
+
+### Step 1: Download or generate a package
+
+If you use Munki, Chef, Puppet, or another configuration management tool, download the client (agent) for your tool. You can find the client on each tool's GitHub or website. For example, you can download Munki, the Munki client on their [releases page on GitHub](https://github.com/munki/munki/releases). 
+
+Make sure the file you download is a `.pkg` file.
+
+If you plan to run a custom script during macOS setup, you'll need to generate a package. The [munkipkg tool](https://github.com/munki/munki-pkg) is a popular tool for generating packages.
+
+Make sure the package you generate is a `.pkg` file.
+
+### Step 2: Sign the package
+
+1. Obtain an appropriate TLS/SSL certificate with signing capability. You can do this by:
+   - Using an [Apple developer account](https://developer.apple.com/account).
+     - [Link your developer account to
+       Xcode](https://help.apple.com/xcode/mac/current/#/dev154b28f09) or [using your developer account](https://developer.apple.com/help/account/create-certificates/create-developer-id-certificates).
+     - Ensure you choose "Developer ID Installer" as the certificate type.
+     - Verify the certificate is saved to your macOS Keychain.
+
+   - Acquiring a certificate from third parties that meet the requirements.
+     - Follow their instructions for creating a TLS/SSL certificate with signing usage.
+     - Add the resulting `.p12` to your keychain.
+2. Sign your `pkg` with a valid Developer ID certificate:\
+\
+```productsign --sign "Developer ID Installer: Your Developer Name (SerialNumber)" /path/to/your.pkg /path/to/your-signed.pkg``` \
```

Comment:
@spokanemac is there a UI option for signing? Just curious.

Reply:
Gotcha. Nice. Yeah, I think starting with the productsign utility makes sense. Looks like it's include on Macs by default.